libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -99,14 +101,35 @@ string IRGenerator::generate(ContractDefinition const& _contract)
 
 	t("CreationObject", creationObjectName(_contract));
 	t("memoryInit", memoryInit());
-	t("constructor", constructorCode(_contract));
+
+	FunctionDefinition const* constructor = _contract.constructor();
+	t("callValueCheck", !constructor || !constructor->isPayable() ? callValueCheck() : "");
+	string constructorParams;
+	if (constructor && !constructor->parameters().empty())
+	{
+		constructorParams = suffixedVariableNameList(
+			"param_" + m_context.functionName(*constructor) + "_",
```

Comment:
Do we need the more complicated name?

Reply:
It is now out of function, I added it for safety, but can simplify if you think it is not needed

Reply:
You are right, this is the global context. In that case, maybe even use variables names known to be unique (`newYulVariable`)?

Reply:
Correct, will do like that